geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/metadata/FileBasedMetadata.scala

```diff
@@ -52,37 +58,61 @@ import scala.util.control.NonFatal
   * @param leafStorage leaf storage
   */
 class FileBasedMetadata(
-    fc: FileContext,
-    directory: Path,
+    private val fc: FileContext,
+    val directory: Path,
     val sft: SimpleFeatureType,
     val encoding: String,
     val scheme: PartitionScheme,
     val leafStorage: Boolean
-  ) extends StorageMetadata {
+  ) extends StorageMetadata with MethodProfiling with LazyLogging {
+
+  import FileBasedMetadata._
 
   import scala.collection.JavaConverters._
 
-  private val partitions = new ConcurrentHashMap[String, PartitionMetadata]()
+  private val expiry = PathCache.CacheDurationProperty.toDuration.get.toMillis
 
-  override def getPartition(name: String): Option[PartitionMetadata] = Option(partitions.get(name))
+  // cache of files associated with each partition
+  // we use a cache to provide lazy non-blocking refresh, but the cache will only ever have 1 element in it
+  private val partitions: LoadingCache[BoxedUnit, ConcurrentMap[String, PartitionFiles]] =
+    Caffeine.newBuilder().refreshAfterWrite(expiry, TimeUnit.MILLISECONDS).build(
+      new CacheLoader[BoxedUnit, ConcurrentMap[String, PartitionFiles]]() {
+        override def load(key: BoxedUnit): ConcurrentMap[String, PartitionFiles] = readPartitionFiles(8)
```

Comment:
Oh man, BoxedUnit as a cache's key.  This is neat and simultaneously, I have to try and think of alternatives.  

I don't have any at the minute.:)